generatedocs.py

```diff
@@ -0,0 +1,413 @@
+def assetsInPathRecursive(root):
+    filenames = []
+
+    for path, subdirs, files in os.walk(root):
+        for file in files:
+            if file.endswith(".asset"):
+                filenames.append(os.path.join(path, file))
+
+    return filenames 
+
+# Find all .assets in directory - no subdirs
+def assetsInPath(path):    
+    filenames = [filename for filename in os.listdir(path) if filename.endswith(".asset")]
+    # Prepend the path to the filename
+    return list(map(lambda filename: os.path.join(path, filename), filenames))
+
+# Matches an asset component name with the words in a file
+# Returns the content of the file and the lines where the 
+# matches occured, as well as the header if specified
+def getLinesAndContentFromFile(assetFile, regex, lookForHeader = False):
+    lines = []
+    isHeaderComment = True
+    header = ""
+    content = ""
+    description = ""
+    luaComment = "-- "
+    headerFinished = 0
+    with open(assetFile, 'r', encoding='utf8') as file:
+        if not file.readable():
+            return None
+        # Read file line by line
+        for l_no, line in enumerate(file, 1):
+            # Check for header comment at top of file 
+            if not line.startswith(luaComment):
+                isHeaderComment = False
+            
+            # If we are in header comment, split into header 
+            # and description
+            if lookForHeader and isHeaderComment:
+                comment = line.split(luaComment)[1]
+                if l_no == 1:
+                    header = comment
+                else:
+                    description += comment
+                    headerFinished = l_no + 1
+            # Else, get the content of the example
+            else:
+                content += line
+                # Search for the regex pattern
+                if re.search(regex, line):
+                    # If the header has been removed we need to adjust the line number
+                    lines.append(l_no - headerFinished)
+    # If there were any matches to regex, set the content as the example
+    if len(lines) > 0:
+        return { "header" : header, "description": description, "content" : content, "lines" : lines }
+    else: 
+        return None
+
+# Takes an asset component name and matches it to all files in a 
+# path, and return the shortest asset with matches.
+def findShortestAssetInPath(path, name):
+    assetFiles = assetsInPathRecursive(path)
+    # Sort by shortest asset file first                
+    # This will ensure the simplest asset is displayed
+    assetFiles.sort(key=getFileLength)
+
+    # Find example for the asset component
+    for assetFile in assetFiles:   
+        # Search for Type = "<name>"
+        regex = r'Type = \"' + name + r'\"'
+        example = getLinesAndContentFromFile(assetFile, regex)
+        # As soon as we have a match, we return it
+        # It will be the shortest assets since they are sorted on length
```

Comment:
Coments here seems a bit wordy for such a short function. 

```suggestion
    # Find first example matching the asset component (files are sorted by length)
    for assetFile in assetFiles:   
        # Search for Type = "<name>"
        regex = r'Type = \"' + name + r'\"'
        example = getLinesAndContentFromFile(assetFile, regex)
```